### PR TITLE
Change the (optional) sentence splitting pipe to use pysbd

### PR DESCRIPTION
**Pull request by danielkingai2**

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -18,3 +18,6 @@ mypy
 
 # Required for releases.
 twine
+
+# required for the tests to run, or to use the custom sentence splitter
```
Comment (DeNeutoy) [inline requirements.in]: This needs to be added to `setup.py` too 👍

File: scispacy/custom_sentence_segmenter.py [hunk 2]
```diff
@@ -1,122 +1,51 @@
...
-        doc[open_bracket.i].is_sent_start = False
+        if token.text.replace('\n', '') == '':
+            token.is_sent_start = False
+        elif len(built_up_sentence) >= len(current_segment):
+            token.is_sent_start = True
+
+            # handle the rare (impossible?) case where spacy tokenizes over a sentence boundary that
+            # pysbd finds
+            built_up_sentence = ' '*int(len(built_up_sentence) - len(current_segment))
+            built_up_sentence = token.string
```
Comment (DeNeutoy) [inline scispacy/custom_sentence_segmenter.py]: just to double check, `token.string` includes all post whitespace, right? e.g it is the same as `token.text_with_ws`

File: scispacy/custom_sentence_segmenter.py [hunk 3]
```diff
@@ -1,122 +1,51 @@
...
-        prev_tokens = prev_tokens[1:] + [token]
-
-    # any unmatched parens should not be the start of a sentence
-    for open_paren in parens_stack:
-        doc[open_paren.i].is_sent_start = False
-
-    # any unmatched brackets should not be the start of a sentence
-    for open_bracket in brackets_stack:
-        doc[open_bracket.i].is_sent_start = False
+        if token.text.replace('\n', '') == '':
```
Comment (DeNeutoy) [inline scispacy/custom_sentence_segmenter.py]: Is this true in spacy? Maybe good to add a note to the docstring to indicate that newlines cannot be EOS tokens. Might also be worth checking for windows newlines, like `\r` etc.

File: scispacy/custom_sentence_segmenter.py (hunk 2 above)
Comment (danielkingai2) [reply scispacy/custom_sentence_segmenter.py]: yes, but you are correct that the documentation says to use text_with_ws, so ill switch to that.  This also made me realize that I need to special case whitespace at the beginning of a string

File: scispacy/custom_sentence_segmenter.py (hunk 3 above)
Comment (danielkingai2) [reply scispacy/custom_sentence_segmenter.py]: yeah, 
```
>>> doc = nlp("With a\nnew line")
>>> list(doc)
[With, a, 
, new, line]
>>> [token.text.replace('\n', '') for token in doc]
['With', 'a', '', 'new', 'line']
```
I'll add \r as well. Seems pysbd just drops both of those from its output

**Pull request merged**